Replace Lpsolve with Coin-OR/SYMPHONY in placer

srini229 opened the pull request.

stevenmburns commented on PlaceRouteHierFlow/thirdparty/SymCMakeFiles/CMakeLists.Cgl:
```diff
@@ -0,0 +1,274 @@
+get_ac_init_version(${Cgl_DIR}/configure.ac CGL)
```
@srini229 We shouldn't put these files in our ALIGN-public repo because they have the wrong license. We certainly can't have them in the repo and then change something. 
We need to have these in a different repo. We could add a symbolic link during our build flow.

srini229 replied:
@stevenmburns Can I create a repo under the https://github.com/ALIGN-analoglayout hierarchy parallel to ALIGN-public to host these CMakeFiles?

stevenmburns commented:
@srini229 We need to make sure the LGPL/Eclipse licensed code (even Cmakefile) are not in our ALIGN-public repo.
These should be moved to a parallel repository. Let me know if you need help setting one up.

Don't merge this with master until this is resolved.

srini229 commented:
> @srini229 We need to make sure the LGPL/Eclipse licensed code (even Cmakefile) are not in our ALIGN-public repo. These should be moved to a parallel repository. Let me know if you need help setting one up.
> 
> Don't merge this with master until this is resolved.

@stevenmburns I changed the build flow to call configure/make using ExternalProject inside our cmake flow instead of using the external CMakeFiles that has the LGPL license. This way we can avoid another dependency/repo. This works for all the three wheels we support now. @parijatm can comment if this way of building is okay. 

I also added an option `--ilp_solver` to choose from `{symphony, lpsolve}` at the toplevel "schematic2layout.py" so that we can just change defaults if the flow breaks for some reason with symphony.

stevenmburns commented on PlaceRouteHierFlow/thirdparty/symphony.cmake:
```diff
@@ -3,121 +3,77 @@ FetchContent_Declare(
-  target_include_directories(libOsiClp INTERFACE ${symphony_SOURCE_DIR}/Clp/src/OsiClp)
-  target_include_directories(libOsiSym INTERFACE ${symphony_SOURCE_DIR}/SYMPHONY/src/OsiSym)
-  target_include_directories(libCoinUtils INTERFACE ${symphony_SOURCE_DIR}/CoinUtils/src)
+  message(STATUS ${MAKE})
```
This looks like a great way to do it.

stevenmburns commented on PlaceRouteHierFlow/placer/PlacerIfc.h:
```diff
@@ -11,7 +11,7 @@ class PlacerIfc {
   public:
   PlacerIfc(PnRDB::hierNode& currentNode, int numLayout, string opath, int effort, PnRDB::Drc_info& drcInfo, const PlacerHyperparameters& hyper,
-            bool select_in_ILP);
+            bool select_in_ILP, int ilp_solver);
```
The hyper parameters is probably a better place to put this (and also select_in_ILP), but this is no big deal.

stevenmburns commented:
This is probably ready to go. I'll wait until after the Thursday meeting to sign off on it.

stevenmburns reviewed (approved):
Looks good. Thanks for taking the time to solve the licensing issues.
